### PR TITLE
BEI-21: Created the Sequelize migration and model files for the recip…

### PR DESCRIPTION
**Pull request by LucianaDera**
WHY THIS PR?
 We need a Sequelize migration and model files to define the structure for the recipes table.
 
WHAT DOES THIS PR DO?
 - This step allows Sequelize to recognize the model and map the recipe to the respective database table.

HOW CAN THIS PR BE TESTED?
N/A

THINGS YOU MAY WANT THE REVIEWER TO NOTE.
N/A

### CODE REVIEW

File: src/core/database/migrations/20230728072728-create-recipes.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        allowNull: false,
+        references: {
+          model: 'users',
+          key: 'id',
+        },
+      },
+      slug: {
+        type: Sequelize.STRING(50),
+        allowNull: false,
+      },
```
Comment (RICHYDOS) [inline src/core/database/migrations/20230728072728-create-recipes.js:23]: Please add unique property here

File: src/core/models/recipe.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    userId: {
+      type: DataTypes.INTEGER.UNSIGNED,
+      autoIncrement: true,
+      primaryKey: true,
+      allowNull: false
+    },
+    slug: {
+      type: DataTypes.STRING(50),
+      allowNull: false,
+    },
```
Comment (RICHYDOS) [inline src/core/models/recipe.ts:35]: Add unique property here too

File: src/core/database/migrations/20230728072728-create-recipes.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    await queryInterface.createTable('recipes', {
+      id: {
+        type: Sequelize.INTEGER(11).UNSIGNED,
+        autoIncrement: true,
+        primaryKey: true,
+        allowNull: false,
+      },
+      userId: {
```
Comment (hizikex) [inline src/core/database/migrations/20230728072728-create-recipes.js:12]: I thought this should be snake case: something like this user_id

File: src/core/interfaces/recipes.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export interface RecipeAttributes {
+ id: number;
+ userId:number;
+ slug:string;
+ title: string;
+ description:string;
+ image:string;
```
Comment (RICHYDOS) [inline src/core/interfaces/recipes.ts:7]: Please show that image is optional in the interface

**Review by RICHYDOS — CHANGES_REQUESTED** [review]
Well done

File: src/core/database/migrations/20230728072728-create-recipes.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      },
+      image: {
+        type: Sequelize.STRING,
+        allowNull: true,
+      },
+      servings: {
+        type: Sequelize.STRING(225),
+        allowNull: false,
+      },
+      durationInMinutes: {
```
Comment (hizikex) [inline src/core/database/migrations/20230728072728-create-recipes.js:40]: This should be in snake case too.

File: src/core/interfaces/recipes.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export interface RecipeAttributes {
+ id: number;
+ userId:number;
```
Comment (hizikex) [inline src/core/interfaces/recipes.ts:3]: reflect the snake case here

File: src/core/interfaces/recipes.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+ slug:string;
+ title: string;
+ description:string;
+ image:string;
+ servings:string;
+ durationInMinutes:number
```
Comment (hizikex) [inline src/core/interfaces/recipes.ts:9]: reflect the snake case here.

File: src/core/models/recipe.ts (hunk 2 above)
Comment (hizikex) [inline src/core/models/recipe.ts:35]: slug should be unique

**Comment by norb-c** [comment]
> Some files could not be reviewed due to errors:

Can we fix this hound CI errors

File: src/core/models/recipe.ts [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    }
+  },
+  {
+    sequelize,
+    tableName: 'recipes'
+  }
+);
+
+Recipe.belongsTo(User, {
+  foreignKey: 'userId',
```
Comment (norb-c) [inline src/core/models/recipe.ts:65]:
```suggestion
  foreignKey: 'user_id',
```

File: src/core/models/recipe.ts [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+  },
+  image:{
+    type: DataTypes.STRING,
+    allowNull: true,
+  },
+  servings:{ 
+    type: DataTypes.STRING(225),
+    allowNull: false
+  },
+  durationInMinutes:{
```
Comment (norb-c) [inline src/core/models/recipe.ts:53]:
```suggestion
  durationIn_minutes:{ 
```

File: src/core/models/recipe.ts [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+
+export class Recipe extends Model<InferAttributes<Recipe>, InferCreationAttributes<Recipe, { omit: 'id' }>> implements RecipeAttributes{
+  declare id: CreationOptional<number>;
+  declare user_id:number;
+  declare slug:string;
+  declare title: string;
+  declare description:string;
+  declare image:string;
+  declare servings:string;
```
Comment (hizikex) [inline src/core/models/recipe.ts:14]: servings: number;

File: src/core/models/recipe.ts [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+    allowNull: false
+  },
+  image:{
+    type: DataTypes.STRING,
+    allowNull: true,
+  },
+  servings:{ 
+    type: DataTypes.STRING(225),
+    allowNull: false
+  },
```
Comment (hizikex) [inline src/core/models/recipe.ts:52]: this datatype is an integer

File: src/core/database/migrations/20230728072728-create-recipes.js [hunk 12]
```diff
@@ -38,7 +38,7 @@ module.exports = {
         type: Sequelize.STRING(225),
         allowNull: false,
       },
-      durationInMinutes: {
+      durationIn_minutes: {
```
Comment (hizikex) [inline src/core/database/migrations/20230728072728-create-recipes.js:41]: make this look like this duration_in_minutes

File: src/core/interfaces/recipes.ts [hunk 13]
```diff
@@ -5,6 +5,6 @@ export interface RecipeAttributes {
  title: string;
  description:string;
  image?:string;
- servings:string;
- durationInMinutes:number
+ servings:number;
```
Comment (hizikex) [inline src/core/interfaces/recipes.ts]: put a space between all the keys and the datatype value in the interface.

File: src/core/database/migrations/20230728072728-create-recipes.js (hunk 3 above)
Comment (RICHYDOS) [reply src/core/database/migrations/20230728072728-create-recipes.js:12]: @LucianaDera pls look at hizikex comment here

**Review by RICHYDOS — CHANGES_REQUESTED** [review]

File: src/core/models/recipe.ts [hunk 14]
```diff
@@ -47,10 +47,10 @@ Recipe.init(
     allowNull: true,
   },
   servings:{ 
-    type: DataTypes.STRING(225),
+    type: DataTypes.INTEGER,
     allowNull: false
   },
-  durationInMinutes:{ 
+  durationIn_minutes:{
```
Comment (hizikex) [inline src/core/models/recipe.ts]: make it like "duration_in_minutes"

File: src/core/models/recipeIngredient.ts [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+    },
+    quantity: {
+      type: DataTypes.DECIMAL,
+      allowNull: false
+    },
+  unit: {
+    type: DataTypes.STRING,
+    allowNull: false
+  },
+  },
```
Comment (hizikex) [inline src/core/models/recipeIngredient.ts:40]: line 36-39 is not aligned

Comment (LucianaDera) [reply src/core/models/recipeIngredient.ts]: The file is not suppose to be there

**Review by hizikex — APPROVED** [review]

**Review by RICHYDOS — APPROVED** [review]

**Pull request merged**